lib/server/auth.js

```diff
@@ -114,48 +115,75 @@ class DataAccess extends EventEmitter {
     });
     LOGGER.debug('ACCESS GROUP IDS: ' + agroupIds.join(' '));
 
-    let authurl = config.provide().get('authurl');
+    let auth_server = config.provide().get('auth_server');
+    let authurl = auth_server.url;
     let reqbody = {groups: agroupIds};
-    let reqopts = url.parse(authurl);
-    reqopts.method = 'POST';
+    let reqopts = {};
+    reqopts.url     = url.parse(authurl);
+    reqopts.method  = 'POST';
+    reqopts.headers = {
+      'Content-Type': 'application/json'
+    };
+
+    if ( authurl.startsWith('https://') ) {
+      let auth_cert = auth_server.client_cert;
+      if ( auth_cert ) {
+        reqopts.cert = auth_cert;
+      }
```

Comment:
Nit: Couldn't we just do `reqopts.cert = auth_server.client_cert;`? (And similar for below)